lib/watir-screenshot-stitch.rb

```diff
@@ -52,20 +52,31 @@ def base64_geckodriver
     #   @param [Watir::Browser] browser
     # @param [Hash] opts
     #
+    # @return [Thread]
```

Comment:
I think the point of the yard documentation is that it's not obvious. It doesn't make sense for the class to store an array of threads with the current implementation since there's only one active thread per instance and for the actually stitching there's only one active thread for the class. Again I could see a possible improvement where instead of trying to free up watir to continue we were actually trying to write optimally threaded code. My goal here isn't to make this as fast as possible but rather to eliminate an extremely slow bottleneck. Allowing a number of threads would be useful in the narrow use case where someone is taking many screenshot stitches in quick succession. It would be nice to use threading to speed that use case up but in my experience unless your testing locally by the time the next page is loaded and it makes sense to take another full page screenshot the previous stitch is complete anyway.

Reply:
I'm not saying I've never done it. :) I'm just saying that it is generally bad practice to have a return value that isn't expected (Principle of Least Surprise and all that). Especially so when it is an existing method changing the return value (agreed less of an issue when going from nil to an instance).

> there's only one active thread per instance
Ok, but the created threads can be stored stored in any instance variable, so long as you don't create multiple instances of the class, you have references to each thread regardless of which thread the instance is actually in.